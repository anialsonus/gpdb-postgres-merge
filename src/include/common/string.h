/*
 *	string.h
 *		string handling helpers
 *
<<<<<<< HEAD
 *  Portions Copyright (c) 1996-2015, PostgreSQL Global Development Group
 *  Portions Copyright (c) 1994, Regents of the University of California
=======
 *	Portions Copyright (c) 1996-2015, PostgreSQL Global Development Group
 *	Portions Copyright (c) 1994, Regents of the University of California
>>>>>>> ab93f90cd3a4fcdd891cee9478941c3cc65795b8
 *
 *	src/include/common/string.h
 */
#ifndef COMMON_STRING_H
#define COMMON_STRING_H

extern bool pg_str_endswith(const char *str, const char *end);

#endif   /* COMMON_STRING_H */
