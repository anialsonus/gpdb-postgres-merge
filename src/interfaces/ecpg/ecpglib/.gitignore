<<<<<<< HEAD
libecpg.a
libecpg.so
libecpg.so.5
path.c
thread.c
libecpg.so.5.2
strlcpy.c
exports.list
pgstrcasecmp.c
libecpg.so.6
libecpg.so.6.1
=======
/libecpgdll.def
/libecpgddll.def
/blibecpgdll.def
/exports.list

/path.c
/pgstrcasecmp.c
/strlcpy.c
/thread.c
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
