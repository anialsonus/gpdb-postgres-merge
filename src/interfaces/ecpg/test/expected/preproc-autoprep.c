/* Processed by ecpg (regression mode) */
/* These include files are added by the preprocessor */
<<<<<<< HEAD
=======
#include <ecpgtype.h>
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#include <ecpglib.h>
#include <ecpgerrno.h>
#include <sqlca.h>
/* End of automatic include section */
#define ECPGdebug(X,Y) ECPGdebug((X)+100,(Y))

#line 1 "autoprep.pgc"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* test automatic prepare for all statements */

#line 1 "regression.h"






#line 6 "autoprep.pgc"


<<<<<<< HEAD
int main() {
=======
int main(int argc, char* argv[]) {
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
  /* exec sql begin declare section */
  	     
  
#line 10 "autoprep.pgc"
<<<<<<< HEAD
 int item [ 4 ] , ind [ 4 ] , i = 1 ;
=======
 int  item [ 4 ]    ,  ind [ 4 ]    ,  i   = 1 ;
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
/* exec sql end declare section */
#line 11 "autoprep.pgc"


  ECPGdebug(1, stderr);
  { ECPGconnect(__LINE__, 0, "regress1" , NULL, NULL , NULL, 0); }
#line 14 "autoprep.pgc"


  /* exec sql whenever sql_warning  sqlprint ; */
#line 16 "autoprep.pgc"

  /* exec sql whenever sqlerror  sqlprint ; */
#line 17 "autoprep.pgc"


<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "create table T ( Item1 int , Item2 int )", ECPGt_EOIT, ECPGt_EORT);
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "create  table T ( Item1 int   , Item2 int   )    ", ECPGt_EOIT, ECPGt_EORT);
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#line 19 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 19 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 19 "autoprep.pgc"


<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 , null )", ECPGt_EOIT, ECPGt_EORT);
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 , null ) ", ECPGt_EOIT, ECPGt_EORT);
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#line 21 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 21 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 21 "autoprep.pgc"

<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 , $1  )", 
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 ,  $1  ) ", 
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
	ECPGt_int,&(i),(long)1,(long)1,sizeof(int), 
	ECPGt_NO_INDICATOR, NULL , 0L, 0L, 0L, ECPGt_EOIT, ECPGt_EORT);
#line 22 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 22 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 22 "autoprep.pgc"

  i++;
<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 , $1  )", 
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "insert into T values ( 1 ,  $1  ) ", 
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
	ECPGt_int,&(i),(long)1,(long)1,sizeof(int), 
	ECPGt_NO_INDICATOR, NULL , 0L, 0L, 0L, ECPGt_EOIT, ECPGt_EORT);
#line 24 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 24 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 24 "autoprep.pgc"

<<<<<<< HEAD
  { ECPGprepare(__LINE__, NULL, 0, "i", " insert into T values ( 1 , 2 ) ");
=======
  { ECPGprepare(__LINE__, NULL, 0, "i", " insert into T values ( 1 , 2 )  ");
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#line 25 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 25 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 25 "autoprep.pgc"

  { ECPGdo(__LINE__, 0, 1, NULL, 0, 1, "i", ECPGt_EOIT, ECPGt_EORT);
#line 26 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 26 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 26 "autoprep.pgc"


<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "select Item2 from T order by Item2 nulls last", ECPGt_EOIT, 
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "select  Item2  from T    order by Item2  nulls last", ECPGt_EOIT, 
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
	ECPGt_int,(item),(long)1,(long)4,sizeof(int), 
	ECPGt_int,(ind),(long)1,(long)4,sizeof(int), ECPGt_EORT);
#line 28 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 28 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 28 "autoprep.pgc"


  for (i=0; i<4; i++)
  	printf("item[%d] = %d\n", i, ind[i] ? -1 : item[i]);

<<<<<<< HEAD
  /* declare C cursor for select Item1 from T */
#line 33 "autoprep.pgc"


  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "declare C cursor for select Item1 from T", ECPGt_EOIT, ECPGt_EORT);
=======
  /* declare C  cursor  for select  Item1  from T    */
#line 33 "autoprep.pgc"


  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "declare C  cursor  for select  Item1  from T   ", ECPGt_EOIT, ECPGt_EORT);
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#line 35 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 35 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 35 "autoprep.pgc"


  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "fetch 1 in C", ECPGt_EOIT, 
	ECPGt_int,&(i),(long)1,(long)1,sizeof(int), 
	ECPGt_NO_INDICATOR, NULL , 0L, 0L, 0L, ECPGt_EORT);
#line 37 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 37 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 37 "autoprep.pgc"

  printf("i = %d\n", i);

  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "close C", ECPGt_EOIT, ECPGt_EORT);
#line 40 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 40 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 40 "autoprep.pgc"


<<<<<<< HEAD
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "drop table T", ECPGt_EOIT, ECPGt_EORT);
=======
  { ECPGdo(__LINE__, 0, 1, NULL, 0, ECPGst_prepnormal, "drop table T ", ECPGt_EOIT, ECPGt_EORT);
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
#line 42 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 42 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 42 "autoprep.pgc"


  { ECPGdisconnect(__LINE__, "ALL");
#line 44 "autoprep.pgc"

if (sqlca.sqlwarn[0] == 'W') sqlprint();
#line 44 "autoprep.pgc"

if (sqlca.sqlcode < 0) sqlprint();}
#line 44 "autoprep.pgc"


  return 0;
}
