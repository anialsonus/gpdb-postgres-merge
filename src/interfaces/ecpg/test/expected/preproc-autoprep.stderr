[NO_PID]: ECPGdebug: set to 1
[NO_PID]: sqlca: code: 0, state: 00000
<<<<<<< HEAD
[NO_PID]: ECPGconnect: opening database regress1 on <DEFAULT> port <DEFAULT>  
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 19: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 19: name ecpg1; query: "create table T ( Item1 int , Item2 int )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 19: query: create table T ( Item1 int , Item2 int ); with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 19: using PQexecPrepared for "create table T ( Item1 int , Item2 int )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 19: OK: CREATE TABLE
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 21: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 21: name ecpg2; query: "insert into T values ( 1 , null )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 21: query: insert into T values ( 1 , null ); with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 21: using PQexecPrepared for "insert into T values ( 1 , null )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 21: OK: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 22: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 22: name ecpg3; query: "insert into T values ( 1 , $1  )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 22: query: insert into T values ( 1 , $1  ); with 1 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 22: using PQexecPrepared for "insert into T values ( 1 , $1  )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: free_params on line 22: parameter 1 = 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 22: OK: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 24: statement found in cache; entry 1640
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 24: query: insert into T values ( 1 , $1  ); with 1 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 24: using PQexecPrepared for "insert into T values ( 1 , $1  )"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: free_params on line 24: parameter 1 = 2
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 24: OK: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 25: name i; query: " insert into T values ( 1 , 2 ) "
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 26: query:  insert into T values ( 1 , 2 ) ; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 26: using PQexecPrepared for " insert into T values ( 1 , 2 ) "
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 26: OK: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 28: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 28: name ecpg4; query: "select Item2 from T order by Item2 nulls last"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 28: query: select Item2 from T order by Item2 nulls last; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 28: using PQexecPrepared for "select Item2 from T order by Item2 nulls last"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 28: correctly got 4 tuples with 1 fields
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data on line 28: RESULT: 1 offset: -1; array: yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data on line 28: RESULT: 2 offset: -1; array: yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data on line 28: RESULT: 2 offset: -1; array: yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data on line 28: RESULT:  offset: -1; array: yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 35: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 35: name ecpg5; query: "declare C cursor for select Item1 from T"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 35: query: declare C cursor for select Item1 from T; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 35: using PQexecPrepared for "declare C cursor for select Item1 from T"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 35: OK: DECLARE CURSOR
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 37: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 37: name ecpg6; query: "fetch 1 in C"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 37: query: fetch 1 in C; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 37: using PQexecPrepared for "fetch 1 in C"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 37: correctly got 1 tuples with 1 fields
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data on line 37: RESULT: 1 offset: -1; array: yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 40: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 40: name ecpg7; query: "close C"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 40: query: close C; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 40: using PQexecPrepared for "close C"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 40: OK: CLOSE CURSOR
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare on line 42: statement not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare on line 42: name ecpg8; query: "drop table T"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 42: query: drop table T; with 0 parameter(s) on connection regress1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 42: using PQexecPrepared for "drop table T"
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute on line 42: OK: DROP TABLE
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg8
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg7
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg6
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg5
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg4
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name i
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg3
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg2
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate on line 0: name ecpg1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_finish: connection regress1 closed
=======
[NO_PID]: ECPGconnect: opening database regress1 on <DEFAULT> port <DEFAULT> 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 19: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 19: NAME: ecpg1 QUERY: create  table T ( Item1 int   , Item2 int   )    
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 19: QUERY: create  table T ( Item1 int   , Item2 int   )     with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 19: using PQexecPrepared for create  table T ( Item1 int   , Item2 int   )    
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 19 Ok: CREATE TABLE
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 21: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 21: NAME: ecpg2 QUERY: insert into T values ( 1 , null ) 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 21: QUERY: insert into T values ( 1 , null )  with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 21: using PQexecPrepared for insert into T values ( 1 , null ) 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 21 Ok: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 22: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 22: NAME: ecpg3 QUERY: insert into T values ( 1 ,  $1  ) 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 22: QUERY: insert into T values ( 1 ,  $1  )  with 1 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 22: using PQexecPrepared for insert into T values ( 1 ,  $1  ) 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: free_params line 22: parameter 1 = 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 22 Ok: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 24: stmt found in cache, entry 6248
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 24: QUERY: insert into T values ( 1 ,  $1  )  with 1 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 24: using PQexecPrepared for insert into T values ( 1 ,  $1  ) 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: free_params line 24: parameter 1 = 2
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 24 Ok: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 25: NAME: i QUERY:  insert into T values ( 1 , 2 )  
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 26: QUERY:  insert into T values ( 1 , 2 )   with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 26: using PQexecPrepared for  insert into T values ( 1 , 2 )  
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 26 Ok: INSERT 0 1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 28: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 28: NAME: ecpg4 QUERY: select  Item2  from T    order by Item2  nulls last
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 28: QUERY: select  Item2  from T    order by Item2  nulls last with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 28: using PQexecPrepared for select  Item2  from T    order by Item2  nulls last
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 28: Correctly got 4 tuples with 1 fields
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data line 28: RESULT: 1 offset: -1 array: Yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data line 28: RESULT: 2 offset: -1 array: Yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data line 28: RESULT: 2 offset: -1 array: Yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data line 28: RESULT:  offset: -1 array: Yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 35: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 35: NAME: ecpg5 QUERY: declare C  cursor  for select  Item1  from T   
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 35: QUERY: declare C  cursor  for select  Item1  from T    with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 35: using PQexecPrepared for declare C  cursor  for select  Item1  from T   
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 35 Ok: DECLARE CURSOR
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 37: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 37: NAME: ecpg6 QUERY: fetch 1 in C
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 37: QUERY: fetch 1 in C with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 37: using PQexecPrepared for fetch 1 in C
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 37: Correctly got 1 tuples with 1 fields
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_get_data line 37: RESULT: 1 offset: -1 array: Yes
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 40: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 40: NAME: ecpg7 QUERY: close C
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 40: QUERY: close C with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 40: using PQexecPrepared for close C
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 40 Ok: CLOSE CURSOR
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_auto_prepare line 42: stmt not in cache; inserting
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGprepare line 42: NAME: ecpg8 QUERY: drop table T 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 42: QUERY: drop table T  with 0 parameter on connection regress1 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 42: using PQexecPrepared for drop table T 
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_execute line 42 Ok: DROP TABLE
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg8
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg7
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg6
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg5
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg4
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: i
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg3
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg2
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ECPGdeallocate line 0: NAME: ecpg1
[NO_PID]: sqlca: code: 0, state: 00000
[NO_PID]: ecpg_finish: Connection regress1 closed.
>>>>>>> 632e7b6353a99dd139b999efce4cb78db9a1e588
[NO_PID]: sqlca: code: 0, state: 00000
